File: src/main/java/ukalus/ui/jcurses/InputField.java
package ukalus.ui.jcurses;

import jcurses.system.InputChar;
import jcurses.widgets.TextField;

/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class InputField extends TextField {

  private boolean done;

  /**
   * Constructor for InputField.
   */
  public InputField() {
    super();
  }

  /**
   * Constructor for InputField.
   * 
   * @param width
   */
  public InputField(int width) {
    super(width);
  }

  /**
   * Constructor for InputField.
   * 
   * @param width
   * @param text
   */
  public InputField(int width, String text) {
    super(width, text);
  }

  /**
   * @see jcurses.widgets.Widget#handleInput(InputChar)
   */
  protected boolean handleInput(InputChar inputChar) {
    int value = inputChar.getCode();

    boolean result = false;
    if (value == '\n') {
      done = true;
      setText(getText(), true);
      result = true;
    } else {
      result = super.handleInput(inputChar);
    }
    return result;
  }

  /**
   * Returns the done.
   * 
   * @return boolean
   */
  public boolean isDone() {
    return done;
  }
}

File: src/main/kotlin/ukalus/ui/jcurses/InputField.kt
package ukalus.ui.jcurses

import jcurses.system.InputChar
import jcurses.widgets.TextField

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class InputField : TextField {

    /**
     * Returns the done.

     * @return boolean
     */
    var isDone: Boolean = false
        private set

    /**
     * Constructor for InputField.
     */
    constructor() : super() {}

    /**
     * Constructor for InputField.

     * @param width
     */
    constructor(width: Int) : super(width) {}

    /**
     * Constructor for InputField.

     * @param width
     * *
     * @param text
     */
    constructor(width: Int, text: String) : super(width, text) {}

    /**
     * @see jcurses.widgets.Widget.handleInput
     */
    override fun handleInput(inputChar: InputChar): Boolean {
        return if (inputChar.code.toChar() == '\n') {
            isDone = true
            setText(text, true)
            true
        } else {
            super.handleInput(inputChar)
        }
    }
}